openmc/statepoint.py

```diff
@@ -94,475 +91,385 @@ def __init__(self, filename):
-        source_sites = self._f['source_bank'].value
+                    if filter_type == 'mesh':
+                        mesh_ids = self._f['tallies/meshes/ids'].value
+                        mesh_keys = self._f['tallies/meshes/keys'].value
 
-        # Initialize SourceSite object for each particle
-        for i in range(self._n_particles):
-            # Initialize new source site
-            site = SourceSite()
+                        key = mesh_keys[mesh_ids == bins][0]
+                        filter.mesh = self.meshes[key]
 
-            # Read position, angle, and energy
-            site._weight, site._xyz, site._uvw, site._E = source_sites[i]
+                    # Add Filter to the Tally
+                    tally.add_filter(filter)
 
-            # Store the source site in the NumPy array
-            self._source[i] = site
+                # Read Nuclide bins
+                nuclide_names = self._f['{0}{1}/nuclides'.format(base, tally_key)].value
 
-    def compute_ci(self, confidence=0.95):
-        """Computes confidence intervals for each Tally bin.
+                # Add all Nuclides to the Tally
+                for name in nuclide_names:
+                    nuclide = openmc.Nuclide(name.decode().strip())
+                    tally.add_nuclide(nuclide)
 
-        This method is equivalent to calling compute_stdev(...) when the
-        confidence is known as opposed to its corresponding t value.
+                # Read score bins
+                n_score_bins = self._f['{0}{1}/n_score_bins'.format(base, tally_key)].value
 
-        Parameters
-        ----------
-        confidence : float, optional
-            Confidence level. Defaults to 0.95.
+                tally.num_score_bins = n_score_bins
 
-        """
+                scores = self._f['{0}{1}/score_bins'.format(
+                    base, tally_key)].value
+                n_user_scores = self._f['{0}{1}/n_user_score_bins'
+                                        .format(base, tally_key)].value
 
-        # Determine significance level and percentile for two-sided CI
-        alpha = 1 - confidence
-        percentile = 1 - alpha/2
+                # Compute and set the filter strides
+                for i in range(n_filters):
+                    filter = tally.filters[i]
+                    filter.stride = n_score_bins * len(nuclide_names)
 
-        # Calculate t-value
-        t_value = scipy.stats.t.ppf(percentile, self._n_realizations - 1)
-        self.compute_stdev(t_value)
+                    for j in range(i+1, n_filters):
+                        filter.stride *= tally.filters[j].num_bins
 
-    def compute_stdev(self, t_value=1.0):
-        """Computes the sample mean and the standard deviation of the mean
-        for each Tally bin.
+                # Read scattering moment order strings (e.g., P3, Y-1,2, etc.)
+                moments = self._f['{0}{1}/moment_orders'.format(
+                    base, tally_key)].value
```

Comment:
Our style guide says follow PEP8 for Python code. PEP8 says continuation lines should match wrapped elements (like for n_user_scores) or use a hanging indent (like here and for score_bins), in which case the indentation level is appropriate (4 spaces). Really, I'm just justifying my lazy use of emacs auto-indent, which I do believe to implement the correct behavior :)